tedana/utils.py

```diff
@@ -384,3 +385,54 @@ def get_resource_path():
     Based on function by Yaroslav Halchenko used in Neurosynth Python package.
     """
     return op.abspath(op.join(op.dirname(__file__), "resources") + op.sep)
+
+
+def get_system_info():
+    """Return information about the system tedana is being run on.
+
+    Returns
+    -------
+    dict
+        Info about system where tedana is run on.
+    """
+    system_info = platform.uname()
+
+    system_dict = {
+        "System": system_info.system,
+        "Node": system_info.node,
+        "Release": system_info.release,
+        "Version": system_info.version,
+        "Machine": system_info.machine,
+        "Processor": system_info.processor,
+    }
+
+    return system_dict
+
+
+def write_tedana_command(arg_vals):
+    """[summary]
+
+    Parameters
+    ----------
+    arg_vals : dict
+        Dictionary with arguments and values of passed to the tedana function.
+
+    Returns
+    -------
+    str
+        Arguments given to tedana and their values formatted in CLI-style.
+    """
+    # Deleta frame from dictionary
+    del arg_vals["frame"]
+
+    command = "tedana"
+
+    for key in arg_vals.keys():
+        # Data and TE are lists
+        if type(arg_vals[key]) is list:
+            values = " ".join(str(v) for v in arg_vals[key])
+        else:
+            values = arg_vals[key]
+        command += f" <b>--{key}</b> {values}"
```

Comment:
You can iterate over both keys and values in Python, which lets you make this a bit neater:
```suggestion
    for k, v in arg_vals:
        # Data and TE are lists
        if type(v) is list:
            values = " ".join(str(x) for x in v)
        else:
            values = v
        command += f" <b>--{k}</b> {values}"
```

Reply:
I didn't know, thanks!